### PR TITLE
[SPARK-7684] [SQL] Invoking HiveContext.newTemporaryConfiguration() shouldn't create new metastore directory

### PR DESCRIPTION
**Pull request by liancheng**
The "Database does not exist" error reported in SPARK-7684 was caused by `HiveContext.newTemporaryConfiguration()`, which always creates a new temporary metastore directory and returns a metastore configuration pointing that directory. This makes `TestHive.reset()` always replaces old temporary metastore with an empty new one.

### CODE REVIEW

**Comment by liancheng** [comment]
cc @marmbrus @yhuai

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala [hunk 1]
```diff
@@ -503,9 +503,10 @@ private[hive] object HiveContext {
   val HIVE_METASTORE_VERSION: String = "spark.sql.hive.metastore.version"
   val HIVE_METASTORE_JARS: String = "spark.sql.hive.metastore.jars"
 
+  private val tempDir = Utils.createTempDir()
+
   /** Constructs a configuration for hive, where the metastore is located in a temp directory. */
   def newTemporaryConfiguration(): Map[String, String] = {
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: Its kind of weird for `newTemporaryConfiguration` to not actually return a new config.  Why not just have a lazy val that is used for the execution conf, and reuse that every time we call reset?

**Comment by liancheng** [comment]
retest this please.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala (hunk 1 above)
Comment (liancheng) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala]: Thanks, updated.

**Comment by liancheng** [comment]
retest this please

**Comment by liancheng** [comment]
I'm merging this to master and branch-1.4.

**Pull request closed without merging**